### PR TITLE
Feature add loading image

### PR DESCRIPTION
**Pull request by alaa-alsalehi**
This PR should adding loader for images based on this issue #867 

this is a Part of [GTC](https://www.facebook.com/groups/gazatechcommunity/) open source initiative

### CODE REVIEW

File: app/src/main/java/com/google/samples/apps/sunflower/adapters/PlantDetailBindingAdapters.kt [hunk 1]
```diff
@@ -32,7 +32,7 @@ fun bindImageFromUrl(view: ImageView, imageUrl: String?) {
         return
     }
     Glide.with(view)
-        .load(imageUrl)
+        .load(imageUrl).placeholder(android.R.drawable.progress_indeterminate_horizontal)
```
Comment (arriolac) [inline app/src/main/java/com/google/samples/apps/sunflower/adapters/PlantDetailBindingAdapters.kt:35]: Nit: formatting:

Glide.with(view)
        .load(imageUrl)
        .placeholder(android.R.drawable.progress_indeterminate_horizontal)

File: app/src/main/java/com/google/samples/apps/sunflower/compose/utils/SunflowerImage.kt [hunk 2]
```diff
@@ -59,6 +64,14 @@ fun SunflowerImage(
         contentScale = contentScale,
         alpha = alpha,
         colorFilter = colorFilter,
-        requestBuilderTransform = requestBuilderTransform
+        requestBuilderTransform = requestBuilderTransform,
+            loading = placeholder {
+                Column(modifier=modifier.fillMaxWidth(),
```
Comment (arriolac) [inline app/src/main/java/com/google/samples/apps/sunflower/compose/utils/SunflowerImage.kt:69]: This should be a `Box` instead

Comment (alaa-alsalehi) [reply app/src/main/java/com/google/samples/apps/sunflower/compose/utils/SunflowerImage.kt:69]: Actually I am not that experience in Compose but when I use Box with the fillMaxWidth like this
```
loading = placeholder {
            Box(modifier.fillMaxWidth()) {
                   CircularProgressIndicator(modifier.align(Alignment.Center))
             }
}
```
I got an ugly loader fill all the widget 
[device-2023-07-01-083420.webm](https://github.com/android/sunflower/assets/1716129/ac923177-e2fb-4710-acc3-fbb39cd382e9)

File: app/src/main/java/com/google/samples/apps/sunflower/adapters/PlantDetailBindingAdapters.kt (hunk 1 above)
Comment (alaa-alsalehi) [reply app/src/main/java/com/google/samples/apps/sunflower/adapters/PlantDetailBindingAdapters.kt]: done

File: app/src/main/java/com/google/samples/apps/sunflower/compose/utils/SunflowerImage.kt (hunk 2 above)
Comment (alaa-alsalehi) [reply app/src/main/java/com/google/samples/apps/sunflower/compose/utils/SunflowerImage.kt]: Now it is done

File: app/src/main/java/com/google/samples/apps/sunflower/compose/utils/SunflowerImage.kt [hunk 3]
```diff
@@ -59,6 +64,12 @@ fun SunflowerImage(
         contentScale = contentScale,
         alpha = alpha,
         colorFilter = colorFilter,
-        requestBuilderTransform = requestBuilderTransform
+        requestBuilderTransform = requestBuilderTransform,
+        loading = placeholder {
+            Box(modifier.fillMaxWidth(), contentAlignment = Alignment.Center) {
+                CircularProgressIndicator(Modifier.size(40.dp)
+                )
```
Comment (arriolac) [inline app/src/main/java/com/google/samples/apps/sunflower/compose/utils/SunflowerImage.kt:71]: Fix formatting - add this a line up

Comment (alaa-alsalehi) [reply app/src/main/java/com/google/samples/apps/sunflower/compose/utils/SunflowerImage.kt]: done

**Review by arriolac — APPROVED** [review]

**Pull request merged**